academy/exchange/cloud/server.py

```diff
@@ -151,7 +189,14 @@ async def _create_mailbox_route(request: Request) -> Response:
             text='Missing or invalid mailbox ID',
         )
 
-    manager.create_mailbox(mailbox_id, behavior)
+    client_id = request.headers.get('client_id', None)
+    try:
+        manager.create_mailbox(client_id, mailbox_id, behavior)
+    except UnauthorizedError:
+        return Response(
+            status=_UNAUTHORIZED_CODE,
```

Comment:
I think I missed this in the last PR, but `_UNAUTHORIZED_CODE` should actually be 401 (right now its 405, which is the method not allowed error code). And in this context I think `_FORBIDDEN_ERROR` (403) is actually the correct code to return (as the user _is_ authorized but doesn't have the right permissions). https://en.wikipedia.org/wiki/List_of_HTTP_status_codes